diff -Nur avs-device-sdk-1.1.0/ACL/test/CMakeLists.txt src/ACL/test/CMakeLists.txt
--- avs-device-sdk-1.1.0/ACL/test/CMakeLists.txt	2017-10-03 06:59:05.000000000 +0800
+++ src/ACL/test/CMakeLists.txt	2017-10-04 00:17:32.422177222 +0800
@@ -3,3 +3,9 @@
 set(LIBRARIES ACL ACLTransportCommonTestLib ${CMAKE_THREAD_LIBS_INIT})
 set(INCLUDE_PATH ${AVSCommon_INCLUDE_DIRS} "${ACL_SOURCE_DIR}/include")
 discover_unit_tests( "${INCLUDE_PATH}" "${LIBRARIES}")
+
+target_link_libraries(AVSConnectionManagerTest atomic)
+target_link_libraries(MessageRouterTest atomic)
+target_link_libraries(MimeParserTest atomic)
+target_link_libraries(HTTP2StreamTest atomic)
+target_link_libraries(HTTP2StreamPoolTest atomic)
\ No newline at end of file
diff -Nur avs-device-sdk-1.1.0/ADSL/test/CMakeLists.txt src/ADSL/test/CMakeLists.txt
--- avs-device-sdk-1.1.0/ADSL/test/CMakeLists.txt	2017-10-03 06:59:05.000000000 +0800
+++ src/ADSL/test/CMakeLists.txt	2017-10-04 00:17:32.422177222 +0800
@@ -6,3 +6,8 @@
 
 set(ADSL_TEST_LIBS ADSL ADSLTestCommon)
 discover_unit_tests("${INCLUDE_PATH}" "${ADSL_TEST_LIBS}")
+
+target_link_libraries(DirectiveRouterTest atomic)
+target_link_libraries(DirectiveSequencerTest atomic)
+target_link_libraries(MessageInterpreterTest atomic)
+target_link_libraries(DirectiveProcessorTest atomic)
\ No newline at end of file
diff -Nur avs-device-sdk-1.1.0/AFML/test/CMakeLists.txt src/AFML/test/CMakeLists.txt
--- avs-device-sdk-1.1.0/AFML/test/CMakeLists.txt	2017-10-03 06:59:05.000000000 +0800
+++ src/AFML/test/CMakeLists.txt	2017-10-04 00:17:32.422177222 +0800
@@ -1 +1,3 @@
 discover_unit_tests("${AFML_SOURCE_DIR}/include" AFML)
+
+target_link_libraries(FocusManagerTest atomic)
diff -Nur avs-device-sdk-1.1.0/AuthDelegate/examples/ExampleAuthDelegateClient/src/CMakeLists.txt src/AuthDelegate/examples/ExampleAuthDelegateClient/src/CMakeLists.txt
--- avs-device-sdk-1.1.0/AuthDelegate/examples/ExampleAuthDelegateClient/src/CMakeLists.txt	2017-10-03 06:59:05.000000000 +0800
+++ src/AuthDelegate/examples/ExampleAuthDelegateClient/src/CMakeLists.txt	2017-10-04 00:17:32.422177222 +0800
@@ -6,4 +6,4 @@
     ${AVSCommon_SOURCE_DIR}/AVS/include
     ${AVSCommon_SOURCE_DIR}/Utils/include)
 target_link_libraries(ExampleAuthDelegateClient
-    AuthDelegate AVSCommon)
+    AuthDelegate AVSCommon atomic)
diff -Nur avs-device-sdk-1.1.0/AuthDelegate/test/CMakeLists.txt src/AuthDelegate/test/CMakeLists.txt
--- avs-device-sdk-1.1.0/AuthDelegate/test/CMakeLists.txt	2017-10-03 06:59:05.000000000 +0800
+++ src/AuthDelegate/test/CMakeLists.txt	2017-10-04 00:17:32.422177222 +0800
@@ -1 +1,3 @@
 discover_unit_tests("${AuthDelegate_SOURCE_DIR}/include" AuthDelegate)
+
+target_link_libraries(AuthDelegateTest atomic)
diff -Nur avs-device-sdk-1.1.0/AVSCommon/AVS/test/CMakeLists.txt src/AVSCommon/AVS/test/CMakeLists.txt
--- avs-device-sdk-1.1.0/AVSCommon/AVS/test/CMakeLists.txt	2017-10-03 06:59:05.000000000 +0800
+++ src/AVSCommon/AVS/test/CMakeLists.txt	2017-10-04 00:17:32.422177222 +0800
@@ -4,3 +4,14 @@
     "${AVSCommon_SOURCE_DIR}/AVS/test"
     "${AVSCommon_SOURCE_DIR}/SDKInterfaces/test")
 discover_unit_tests("${INCLUDE_PATH}" "AVSCommon;AttachmentCommonTestLib")
+
+target_link_libraries(AlexaClientSDKInitTest atomic)
+target_link_libraries(DialogUXStateAggregatorTest atomic)
+target_link_libraries(ExceptionEncounteredSenderTest atomic)
+target_link_libraries(HandlerAndPolicyTest atomic)
+target_link_libraries(CapabilityAgentTest atomic)
+target_link_libraries(NamespaceAndNameTest atomic)
+target_link_libraries(AttachmentWriterTest atomic)
+target_link_libraries(AttachmentManagerV2Test atomic)
+target_link_libraries(AttachmentReaderTest atomic)
+target_link_libraries(AttachmentTest atomic)
diff -Nur avs-device-sdk-1.1.0/AVSCommon/Utils/src/LibcurlUtils.cpp src/AVSCommon/Utils/src/LibcurlUtils.cpp
--- avs-device-sdk-1.1.0/AVSCommon/Utils/src/LibcurlUtils.cpp	2017-10-03 06:59:05.000000000 +0800
+++ src/AVSCommon/Utils/src/LibcurlUtils.cpp	2017-10-04 00:35:14.390232418 +0800
@@ -89,8 +89,13 @@
     }
     if (!(SETOPT(handle, CURLOPT_SSLVERSION, CURL_SSLVERSION_TLSv1_2) &&
           SETOPT(handle, CURLOPT_HTTP_VERSION, CURL_HTTP_VERSION_2_0) &&
+#if defined(SSL_NO_CACERT_CHECK)          
+          SETOPT(handle, CURLOPT_USE_SSL, CURLUSESSL_ALL) && SETOPT(handle, CURLOPT_SSL_VERIFYPEER, 0L) &&
+          SETOPT(handle, CURLOPT_SSL_VERIFYHOST, 0L))) {
+#else
           SETOPT(handle, CURLOPT_USE_SSL, CURLUSESSL_ALL) && SETOPT(handle, CURLOPT_SSL_VERIFYPEER, 1L) &&
           SETOPT(handle, CURLOPT_SSL_VERIFYHOST, 2L))) {
+#endif
         return false;
     }
 
diff -Nur avs-device-sdk-1.1.0/AVSCommon/Utils/test/CMakeLists.txt src/AVSCommon/Utils/test/CMakeLists.txt
--- avs-device-sdk-1.1.0/AVSCommon/Utils/test/CMakeLists.txt	2017-10-03 06:59:05.000000000 +0800
+++ src/AVSCommon/Utils/test/CMakeLists.txt	2017-10-04 00:17:32.422177222 +0800
@@ -1 +1,14 @@
 discover_unit_tests("${AVSCommon_INCLUDE_DIRS}" AVSCommon)
+
+target_link_libraries(ConfigurationNodeTest atomic)
+target_link_libraries(SharedDataStreamTest atomic)
+target_link_libraries(JSONUtilTest atomic)          
+target_link_libraries(TaskQueueTest atomic)
+target_link_libraries(LogEntryStreamTest atomic)    
+target_link_libraries(TaskThreadTest atomic)
+target_link_libraries(ConfigurationNodeTest atomic) 
+target_link_libraries(LoggerTest atomic)            
+target_link_libraries(TimerTest atomic)
+target_link_libraries(ExecutorTest atomic)          
+target_link_libraries(RequiresShutdownTest atomic)  
+target_link_libraries(UUIDGenerationTest atomic)
diff -Nur avs-device-sdk-1.1.0/CapabilityAgents/AIP/test/CMakeLists.txt src/CapabilityAgents/AIP/test/CMakeLists.txt
--- avs-device-sdk-1.1.0/CapabilityAgents/AIP/test/CMakeLists.txt	2017-10-03 06:59:05.000000000 +0800
+++ src/CapabilityAgents/AIP/test/CMakeLists.txt	2017-10-04 00:17:32.422177222 +0800
@@ -6,3 +6,4 @@
     "${AVSCommon_SOURCE_DIR}/AVS/test")
 
 discover_unit_tests("${INCLUDE_PATH}" AIP)
+target_link_libraries(AudioInputProcessorTest atomic)
diff -Nur avs-device-sdk-1.1.0/CapabilityAgents/AudioPlayer/test/CMakeLists.txt src/CapabilityAgents/AudioPlayer/test/CMakeLists.txt
--- avs-device-sdk-1.1.0/CapabilityAgents/AudioPlayer/test/CMakeLists.txt	2017-10-03 06:59:05.000000000 +0800
+++ src/CapabilityAgents/AudioPlayer/test/CMakeLists.txt	2017-10-04 00:55:53.282296810 +0800
@@ -3,3 +3,5 @@
 set(INCLUDE_PATH "${AudioPlayer_INCLUDE_DIRS}" "${AVSCommon_SOURCE_DIR}/SDKInterfaces/test")
 
 discover_unit_tests("${INCLUDE_PATH}" AudioPlayer)
+
+target_link_libraries(AudioPlayerTest atomic)
diff -Nur avs-device-sdk-1.1.0/CapabilityAgents/PlaybackController/test/CMakeLists.txt src/CapabilityAgents/PlaybackController/test/CMakeLists.txt
--- avs-device-sdk-1.1.0/CapabilityAgents/PlaybackController/test/CMakeLists.txt	2017-10-03 06:59:05.000000000 +0800
+++ src/CapabilityAgents/PlaybackController/test/CMakeLists.txt	2017-10-04 00:57:29.586301815 +0800
@@ -7,3 +7,5 @@
     "${AVSCommon_SOURCE_DIR}/SDKInterfaces/test")
 
 discover_unit_tests("${INCLUDE_PATH}" "PlaybackController")
+
+target_link_libraries(PlaybackControllerTest atomic)
diff -Nur avs-device-sdk-1.1.0/CapabilityAgents/Settings/test/CMakeLists.txt src/CapabilityAgents/Settings/test/CMakeLists.txt
--- avs-device-sdk-1.1.0/CapabilityAgents/Settings/test/CMakeLists.txt	2017-10-03 06:59:05.000000000 +0800
+++ src/CapabilityAgents/Settings/test/CMakeLists.txt	2017-10-04 00:59:22.442307681 +0800
@@ -6,4 +6,6 @@
     "${AVSCommon_SOURCE_DIR}/AVS/test"
     "${AVSCommon_SOURCE_DIR}/SDKInterfaces/test")
 
-discover_unit_tests("${INCLUDE_PATH}" "Settings")
\ No newline at end of file
+discover_unit_tests("${INCLUDE_PATH}" "Settings")
+
+target_link_libraries(SettingsTest atomic)
diff -Nur avs-device-sdk-1.1.0/CapabilityAgents/SpeechSynthesizer/test/CMakeLists.txt src/CapabilityAgents/SpeechSynthesizer/test/CMakeLists.txt
--- avs-device-sdk-1.1.0/CapabilityAgents/SpeechSynthesizer/test/CMakeLists.txt	2017-10-03 06:59:05.000000000 +0800
+++ src/CapabilityAgents/SpeechSynthesizer/test/CMakeLists.txt	2017-10-04 00:17:32.422177222 +0800
@@ -3,3 +3,5 @@
 set(INCLUDE_PATH "${SpeechSynthesizer_INCLUDE_DIRS}" "${AVSCommon_SOURCE_DIR}/SDKInterfaces/test")
 
 discover_unit_tests("${INCLUDE_PATH}" SpeechSynthesizer)
+
+target_link_libraries(SpeechSynthesizerTest atomic)
\ No newline at end of file
diff -Nur avs-device-sdk-1.1.0/CapabilityAgents/System/test/CMakeLists.txt src/CapabilityAgents/System/test/CMakeLists.txt
--- avs-device-sdk-1.1.0/CapabilityAgents/System/test/CMakeLists.txt	2017-10-03 06:59:05.000000000 +0800
+++ src/CapabilityAgents/System/test/CMakeLists.txt	2017-10-04 00:43:09.270257100 +0800
@@ -7,3 +7,6 @@
     "${AVSCommon_SOURCE_DIR}/SDKInterfaces/test")
 
 discover_unit_tests("${INCLUDE_PATH}" "AVSSystem;ADSL")
+
+target_link_libraries(EndpointHandlerTest atomic)
+target_link_libraries(UserInactivityMonitorTest atomic)
diff -Nur avs-device-sdk-1.1.0/CertifiedSender/test/CMakeLists.txt src/CertifiedSender/test/CMakeLists.txt
--- avs-device-sdk-1.1.0/CertifiedSender/test/CMakeLists.txt	2017-10-03 06:59:05.000000000 +0800
+++ src/CertifiedSender/test/CMakeLists.txt	2017-10-04 00:49:25.066276632 +0800
@@ -4,4 +4,6 @@
 
 set(TEST_FOLDER "${CertifiedSender_SOURCE_DIR}/test")
 
-discover_unit_tests("${INCLUDE_PATH}" "CertifiedSender" "${TEST_FOLDER}")
\ No newline at end of file
+discover_unit_tests("${INCLUDE_PATH}" "CertifiedSender" "${TEST_FOLDER}")
+
+target_link_libraries(MessageStorageTest atomic)
diff -Nur avs-device-sdk-1.1.0/ContextManager/test/CMakeLists.txt src/ContextManager/test/CMakeLists.txt
--- avs-device-sdk-1.1.0/ContextManager/test/CMakeLists.txt	2017-10-03 06:59:05.000000000 +0800
+++ src/ContextManager/test/CMakeLists.txt	2017-10-04 00:17:32.422177222 +0800
@@ -1 +1,2 @@
 discover_unit_tests("${ContextManager}/include" ContextManager)
+target_link_libraries(ContextManagerTest atomic)
diff -Nur avs-device-sdk-1.1.0/Integration/test/CMakeLists.txt src/Integration/test/CMakeLists.txt
--- avs-device-sdk-1.1.0/Integration/test/CMakeLists.txt	2017-10-03 06:59:05.000000000 +0800
+++ src/Integration/test/CMakeLists.txt	2017-10-04 00:38:24.738242311 +0800
@@ -96,3 +96,12 @@
         message("Skipping build of NetworkIntegrationTests.cpp.")
     endif()
 endif()
+
+target_link_libraries(AlexaAuthorizationDelegateTest atomic)
+target_link_libraries(AlertsIntegrationTest atomic)
+target_link_libraries(AlexaCommunicationsLibraryTest atomic)
+target_link_libraries(AlexaDirectiveSequencerLibraryTest atomic)
+target_link_libraries(AudioInputProcessorIntegrationTest atomic)
+target_link_libraries(AudioPlayerIntegrationTest atomic)
+target_link_libraries(ServerDisconnectIntegrationTest atomic)
+target_link_libraries(SpeechSynthesizerIntegrationTest atomic)
diff -Nur avs-device-sdk-1.1.0/KWD/test/CMakeLists.txt src/KWD/test/CMakeLists.txt
--- avs-device-sdk-1.1.0/KWD/test/CMakeLists.txt	2017-10-03 06:59:05.000000000 +0800
+++ src/KWD/test/CMakeLists.txt	2017-10-04 00:17:32.422177222 +0800
@@ -1 +1,3 @@
-discover_unit_tests("${KWD_SOURCE_DIR}/include" KWD)
\ No newline at end of file
+discover_unit_tests("${KWD_SOURCE_DIR}/include" KWD)
+
+target_link_libraries(AbstractKeywordDetectorTest atomic)
\ No newline at end of file
diff -Nur avs-device-sdk-1.1.0/MediaPlayer/test/CMakeLists.txt src/MediaPlayer/test/CMakeLists.txt
--- avs-device-sdk-1.1.0/MediaPlayer/test/CMakeLists.txt	2017-10-03 06:59:05.000000000 +0800
+++ src/MediaPlayer/test/CMakeLists.txt	2017-10-04 00:52:18.402285641 +0800
@@ -10,3 +10,6 @@
 set(INPUT_FOLDER "${MediaPlayer_SOURCE_DIR}/inputs")
 
 discover_unit_tests("${INCLUDES}" "${LIBRARIES}" "${INPUT_FOLDER}")
+
+target_link_libraries(MediaPlayerTest atomic)
+target_link_libraries(ErrorTypeConversionTest atomic)
diff -Nur avs-device-sdk-1.1.0/PlaylistParser/test/CMakeLists.txt src/PlaylistParser/test/CMakeLists.txt
--- avs-device-sdk-1.1.0/PlaylistParser/test/CMakeLists.txt	2017-10-03 06:59:05.000000000 +0800
+++ src/PlaylistParser/test/CMakeLists.txt	2017-10-04 00:53:20.358288861 +0800
@@ -1,3 +1,5 @@
 cmake_minimum_required(VERSION 3.1 FATAL_ERROR)
 
 discover_unit_tests("${PlaylistParser_SOURCE_DIR}/include" PlaylistParser)
+
+target_link_libraries(PlaylistParserTest atomic)
\ No newline at end of file
diff -Nur avs-device-sdk-1.1.0/SampleApp/src/CMakeLists.txt src/SampleApp/src/CMakeLists.txt
--- avs-device-sdk-1.1.0/SampleApp/src/CMakeLists.txt	2017-10-03 06:59:05.000000000 +0800
+++ src/SampleApp/src/CMakeLists.txt	2017-10-04 00:17:32.430177222 +0800
@@ -43,3 +43,5 @@
 elseif(${CMAKE_SYSTEM_NAME} MATCHES "Windows")
     message(FATAL_ERROR "Windows is unsupported at the moment")  
 endif()
+
+target_link_libraries(SampleApp atomic)
